test/v2/unit/trading.ts

```diff
@@ -0,0 +1,74 @@
+import { coroutine as co } from 'bluebird';
+import * as should from 'should';
+import * as nock from 'nock';
+
+const Wallet = require('../../../src/v2/wallet');
+const TestV2BitGo = require('../../lib/test_bitgo');
+const common = require('../../../src/common');
+
+describe('V2 Trading', function() {
+  const microservicesUri = 'https://bitgo-microservices-endpoint.com';
+  let bitgo;
+  let basecoin;
+  let wallet;
+  let trading;
+  let bgUrl;
+
+  before(co(function *() {
+    bitgo = new TestV2BitGo({ env: 'dev', microservicesUri });
+    bitgo.initializeTestVars();
+    basecoin = bitgo.coin('ofc');
+    basecoin.keychains();
+
+    const walletData = {
+      id: 'walletId',
+      coin: 'tbtc',
+      keys: [
+        'keyid'
+      ]
+    };
+
+    wallet = new Wallet(bitgo, basecoin, walletData);
+    trading = wallet.trading();
+    bgUrl = common.Environments[bitgo.getEnv()].uri;
+  }));
+
+  it('should create and sign a trade payload', co(function *() {
+    const msScope = nock(microservicesUri)
+    .post('/api/trade/v1/payload')
+    .reply(200, {
+      payload: JSON.stringify({
+        walletId: 'walletId',
+        currency: 'tbtc',
+        amount: '100000000',
+        otherParties: ['test_counter_party_1'],
+        nonce: Date.now()
+      })
+    });
+    const platformScope = nock(bgUrl)
+    .get('/api/v2/ofc/key/keyid')
+    .reply(200, {
+      pub: 'xpub1',
+      encryptedPrv: bitgo.encrypt({ input: 'xprv9s21ZrQH143K2MUz7uPUBVzdmvJQE6fPEQCkR3mypPbZgijPqfmGH7pjijdjeJx3oCoxPWVbjC4VYHzgN6wqEfYnnbNjK7jm2CkrvWrvkbR', password: TestV2BitGo.OFC_TEST_PASSWORD })
+    });
+
+    const { payload, signature } = yield trading.signTradePayload({
+      currency: 'tbtc',
+      amount: '100000000',
+      otherParties: ['test_counterparty_1'],
+      walletPassphrase: TestV2BitGo.OFC_TEST_PASSWORD
+    });
+
+    should.exist(payload);
+    // The payload should be a valid JSON object
+    // NOTE: we shouldn't do any more validation than this, as the schema is subject to change often
+    (() => { JSON.parse(payload); }).should.not.throw();
+
+    should.exist(signature);
```

Comment:
Should there be a check that the signature is valid over the payload? (up to you)

Reply:
We don't have a method in the SDK to verify arbitrary signatures, only transactions. If we can implement it without bringing in new dependencies we could consider it.

Reply:
We have `bitcoinjs-message`, which has a `verify()` function.

See https://github.com/BitGo/BitGoJS/blob/6dead153286316484678de2e777afeec9fa5d5fb/src/v2/coins/abstractUtxoCoin.ts#L408